src/ExactOnline.Client.Sdk/Helpers/ApiConnector.cs

```diff
@@ -266,19 +261,19 @@ private Stream GetResponseStream(HttpWebRequest request)
 			}
 
 			Debug.WriteLine("RESPONSE");
+			var response = default(HttpResponseMessage);
 
-			var response = default(WebResponse);
 
 			// Get response. If this fails: Throw the correct Exception (for testability)
 			try
 			{
-				response = request.GetResponse();
-				return response.GetResponseStream();
+				response = _httpClient.SendAsync(request, HttpCompletionOption.ResponseHeadersRead).Result;
+				response.EnsureSuccessStatusCode();
+				return response.Content.ReadAsStreamAsync().Result;

```

Comment:
`.Result` is bad... I know it's sync over async, which is bad in any case... but better to do `.GetAwaiter().GetResult()`. There's a few more places where it's used...